src/identity/caclient.rs

```diff
@@ -218,6 +218,24 @@ pub mod mock {
             state.fetches.push(id.to_owned());
             Ok(certs)
         }
+
+        pub async fn set_error(&mut self, error: bool) {
+            if error {
+                let mut state = self.state.write().await;
+                state.fetches.push(Identity::Spiffe {
```

Comment:
This is a test helper function right? I would suggest moving it into a separate test file or at the very least putting it near the actual tests that use it so that its purpose is clearer

Reply:
I believe [clear_fetches](https://github.com/istio/ztunnel/blob/6d9a457ed533d10b3bc618d7aaa81df8ed2fef7a/src/identity/manager.rs#L819), the function right above this is also a test helper function.

Reply:
I'm not sure if it's a test helper or just a function used outside of tests. clear_fetches looks generally useful

Reply:
The only use of `clear_fetches` I see is in [test_priority in the manager.rs](https://github.com/istio/ztunnel/blob/6d9a457ed533d10b3bc618d7aaa81df8ed2fef7a/src/identity/manager.rs#L819) file.

The way I'm currently setting the error in that `set_error`, I think that if I wanted this function outside of this file I'd have to copy some of the structs like ClientState, ClientConfig and CaClient and the necessary imports over to manager.rs. Is that the suggestion or are you suggesting more simple/elegant?

Reply:
Fair enough; not blocking on it